lib/addons/fluxcd/index.ts

```diff
@@ -9,6 +9,51 @@ import { FluxGitRepository } from "./gitrepository";
 import { FluxKustomization } from "./kustomization";
 import { KubernetesManifest } from 'aws-cdk-lib/aws-eks/lib/k8s-manifest';
 
+/**
+ * Options for a FluxCD repository
+ */
+export interface FluxCDGitRepo extends Omit<spi.GitOpsApplicationDeployment, "repository">{
+
+    /** 
+     * Repository to create source from
+     */
+    repository: Omit<spi.GitRepositoryReference, "path" | "name">
+  /** 
+  * Flux SecretRef */
+  fluxSecretRefName?: string;
+
+  /** 
+  * Internal for Flux sync.
+  * Default `5m0s` */
+  fluxSyncInterval?: string;
+
+    /**
+     * List of kustomizations to create */
+    kustomizations?: FluxKustomizationProps[];
+}
+
+export interface FluxKustomizationProps {
+    /**
+     * Flux Kustomization path within the GitRepository 
+     * Do not use the path in the repository field*/
+    fluxKustomizationPath: string;
+
+    /** 
+    * Flux Kustomization Target Namespace.
+    * Note: when set, this parameter will override all the objects that are part of the Kustomization, please see https://fluxcd.io/flux/components/kustomize/kustomization/#target-namespace */
+    kustomizationTargetNamespace?: string;
+
+    /** 
+    * Flux Kustomization Prune.
+    * Default `true` */
+    fluxPrune?: boolean;
+

```

Comment:
there should be another interval property here for Kustomization. it is different than GitRepository.